### ISSUE TITLE
Is queuing on the worker "processing"?

### ISSUE THREAD

**Issue by max-sixty** [dask/distributed#4939]
**What happened**:

Could I clarify a description in the [documentation](https://distributed.dask.org/en/latest/scheduling-state.html)?

It states:

> Processing: Actively being computed by one or more workers

And also:

> Multiple tasks may be submitted to a worker in advance and the worker will run them eventually, depending on its execution resources (but see Work Stealing).
All the tasks here are in the “processing” state.

Does this mean that a worker has a queue and all tasks in the queue are "processing", even if they're not being computed? 

**What you expected to happen**:

Happy to PR a change to the first definition to something like "Either actively being computed by one or more workers or in a worker queue" if that's more representative.

**Comment by fjetter** [comment]
> Does this mean that a worker has a queue and all tasks in the queue are "processing", even if they're not being computed?

Yes, the current docs are not 100% accurate. If you feel like opening a PR this would be highly appreciated.

From the schedulers POV, `processing` means all dependencies are available and the task is assigned to a worker for compute. When the task will actually be computed is something the scheduler does not know. The scheduler is not able to distinguish between "queued for execution", "fetching dependencies to execute task" or "execute".
While in this state, the task is also a potential subject to work stealing.

In a sense, `processing` means that the cluster is busy computing it and it will finish eventually without further involvement of the scheduler since at this point the workers will figure it out themselves (as long as nothing bad happens). This is why this state, from the POV of the scheduler, is not distinguished further.

The workers do distinguish this a bit further but to reduce communication overhead, not every worker side transition is communicated to the scheduler. For this, see also https://distributed.dask.org/en/latest/worker.html#internal-scheduling
